firebase-storage/src/main/java/com/google/firebase/storage/network/NetworkRequest.java

```diff
@@ -108,40 +106,37 @@ public static String getAuthority() {
    * @return Url for the target REST call in string form.
    */
   @NonNull
-  public static String getDefaultURL(@NonNull Uri gsUri) {
+  public static Uri getDefaultURL(@NonNull Uri gsUri) {
     Preconditions.checkNotNull(gsUri);
-
     String pathWithoutBucket = getPathWithoutBucket(gsUri);
-    return sNetworkRequestUrl
-        + "/b/"
-        + gsUri.getAuthority()
-        + "/o/"
-        + (pathWithoutBucket != null ? Slashes.unSlashize(pathWithoutBucket) : "");
+    Uri.Builder uriBuilder = sNetworkRequestUrl.buildUpon();
+    uriBuilder.appendPath("b");
+    uriBuilder.appendPath(gsUri.getAuthority());
+    uriBuilder.appendPath("o");
+    uriBuilder.appendPath(pathWithoutBucket);
+    return uriBuilder.build();
   }
 
   /**
-   * Returns the path of the object but excludes the bucket name
+   * Returns the decoded path of the object but excludes the bucket name
    *
    * @param gsUri the "gs://" Uri of the blob.
    * @return the path in string form.
    */
-  @Nullable
-  public static String getPathWithoutBucket(@NonNull Uri gsUri) {
-    String path = gsUri.getEncodedPath();
-    if (path != null && path.startsWith("/")) {
-      // this should always be true.
-      path = path.substring(1);
+  private static String getPathWithoutBucket(@NonNull Uri gsUri) {
+    String path = gsUri.getPath();
+    if (path == null) {
+      return "";
```

Comment:
This change allowed me to remove the nullability handling from two callsites.

Reply:
Another option would be to throw an exception. (It seems like if path is null here, nothing good will result. But I don't know the code base... maybe it's legit? In any case, returning "" maintains the existing behaviour, so might be better to just leave it that way unless you're sure this is always an error case.) No action required.